Google Fonts API key invalid?

Hi! A few hours ago our doc builds over at [PyVRP](https://github.com/PyVRP/PyVRP/) started to fail, with this error message:
```
>>>make html --directory=docs
make: Entering directory '/home/runner/work/PyVRP/PyVRP/docs'
Running Sphinx v7.2.6
Copying example notebooks into docs/source/examples/
making output directory... done
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://www.sphinx-doc.org/en/master/objects.inv...
loading intersphinx inventory from https://numpy.org/doc/stable/objects.inv...
loading intersphinx inventory from https://matplotlib.org/stable/objects.inv...
Fetching: https://content-webfonts.googleapis.com/v1/webfonts?key=AIzaSyAa8yy0GdcGPHdtD083HiGGx_S0vMPScDM with {'x-referer': 'https://explorer.apis.google.com'}
Extension error (sphinx_immaterial.google_fonts):
Handler <function _builder_inited at 0x7fc4f79f7b00> for event 'builder-inited' threw an exception (exception: 403 Client Error: Forbidden for url: https://content-webfonts.googleapis.com/v1/webfonts?key=AIzaSyAa8yy0GdcGPHdtD083HiGGx_S0vMPScDM)
make: Leaving directory '/home/runner/work/PyVRP/PyVRP/docs'
make: *** [Makefile:[20](https://github.com/PyVRP/PyVRP/actions/runs/8191559344/job/22405463658#step:10:21): html] Error 2
Error: Process completed with exit code 2.
```
See [this build](https://github.com/PyVRP/PyVRP/actions/runs/8191559344/job/22405463658) for more details. We are using the latest sphinx-immaterial release (0.11.10), and all this worked until a few hours ago.

When I follow the link to the font over at Google myself, I get the following:
<img width="383" alt="image" src="https://github.com/jbms/sphinx-immaterial/assets/16272507/499e6f2f-f178-459e-bebf-0f19eeda441b">

Which now has an error code of 400, not 403 as in the error message I see in our build. I'm unsure if that is the same issue. Do you know what's going on?

--- a/sphinx_immaterial/google_fonts.py
+++ b/sphinx_immaterial/google_fonts.py
@@ -12,7 +12,7 @@ import io
 import json
 import os
 import re
-from typing import Dict, List, Set, Tuple, Optional
+from typing import Dict, List, Set, Tuple, Optional, cast, Any
 import urllib.parse
 
 import sphinx.application
@@ -23,9 +23,6 @@ from .css_and_javascript_bundles import add_global_css
 from .external_resource_cache import get_url, get_cache_dir
 
 logger = sphinx.util.logging.getLogger(__name__)
-
-# From Google Fonts API Explorer
-_GOOGLE_FONTS_API_KEY = "AIzaSyAa8yy0GdcGPHdtD083HiGGx_S0vMPScDM"
 
 # https://stackoverflow.com/questions/25011533/google-font-api-uses-browser-detection-how-to-get-all-font-variations-for-font
 _FONT_FORMAT_USER_AGENT = {
@@ -144,13 +141,13 @@ def add_google_fonts(app: sphinx.application.Sphinx, fonts: List[str]):
             css_futures = []
             # Fetch list of fonts
             font_metadata = json.loads(
-                get_url(
-                    cache_dir,
-                    f"https://content-webfonts.googleapis.com/v1/webfonts?key={_GOOGLE_FONTS_API_KEY}",
-                    headers={"x-referer": "https://explorer.apis.google.com"},
-                ).decode("utf-8")
+                get_url(cache_dir, "https://fonts.google.com/metadata/fonts").decode(
+                    "utf-8"
+                )
             )
-            font_families = {item["family"]: item for item in font_metadata["items"]}
+            font_families = {
+                item["family"]: item for item in font_metadata["familyMetadataList"]
+            }
             for font in fonts:
                 metadata = font_families.get(font)
                 if metadata is None:
@@ -160,7 +157,9 @@ def add_google_fonts(app: sphinx.application.Sphinx, fonts: List[str]):
                         sorted(font_families),
                     )
                     continue
-                for variant in metadata["variants"]:
+                for variant in cast(
+                    Dict[str, Dict[str, Any]], metadata["fonts"]
+                ).keys():
                     css_future_keys.append((font, variant))
                     css_futures.append(fetch_font(font, variant))
             css_content = dict(zip(css_future_keys, await asyncio.gather(*css_futures)))

--- a/sphinx_immaterial/graphviz.py
+++ b/sphinx_immaterial/graphviz.py
@@ -243,7 +243,7 @@ def render_dot_html(
     ttf_font: Optional[str] = None
     if ttf_font_paths and font is not None:
         # can only support the chosen font if cache exists and a Google font is used
-        ttf_font = ttf_font_paths[(font, "regular")]
+        ttf_font = ttf_font_paths[(font, "400")]
 
     code = _replace_resolved_xrefs(node, code)
 

--- a/sphinx_immaterial/graphviz.py
+++ b/sphinx_immaterial/graphviz.py
@@ -242,8 +242,18 @@ def render_dot_html(
     ttf_font_paths = google_fonts.get_ttf_font_paths(self.builder.app)
     ttf_font: Optional[str] = None
     if ttf_font_paths and font is not None:
-        # can only support the chosen font if cache exists and a Google font is used
-        ttf_font = ttf_font_paths[(font, "400")]
+        try:
+            # can only support the chosen font if cache exists and a Google font is used
+            ttf_font = ttf_font_paths[(font, "400")]
+        except KeyError as exc:
+            # weight `400` might not exist for the specified font
+            all_font_keys = [i for i in ttf_font_paths.keys() if i[0] == font]
+            if not all_font_keys:
+                raise FileNotFoundError(
+                    f"Font file for {font} could not be found in cache"
+                ) from exc
+            # just use first weight for the specified font
+            ttf_font = ttf_font_paths[all_font_keys[0]]
 
     code = _replace_resolved_xrefs(node, code)
 

